network/utils.go

```diff
@@ -55,3 +57,10 @@ func FilterPeers(address string, peers []string) (filtered []string) {
 	}
 	return filtered
 }
+
+// GetRandomUnusedPort returns a random unused port
+func GetRandomUnusedPort() int {
+	listener, _ := net.Listen("tcp", ":0")
```

Comment:
It is possible for it to error out should no ports be available on the system. At this point `panic()`'ing might be best.

Also, best to not use `defer` unless there is multiple return paths in a function.

Reply:
A panic seems very aggressive if we're a shared library, could return an error instead and let the application handle accordingly:
```
func GetRandomUnusedPort() (int, error)
```